application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeMessageListener.java

```diff
@@ -41,16 +47,35 @@ private void updateHistory(MessageReceivedEvent event) {
         long messageId = event.getMessageIdLong();
         long authorId = event.getAuthor().getIdLong();
 
-        database.write(context -> context.newRecord(MESSAGE_HISTORY)
-            .setSentAt(Instant.now())
-            .setGuildId(guildId)
-            .setChannelId(channelId)
-            .setMessageId(messageId)
-            .setAuthorId(authorId)
-            .insert());
+        if (canWriteToDB()) {
+            database.write(context -> context.newRecord(MESSAGE_HISTORY)
+                .setSentAt(Instant.now())
+                .setGuildId(guildId)
+                .setChannelId(channelId)
+                .setMessageId(messageId)
+                .setAuthorId(authorId)
+                .insert());
+
+            incrementRecordsCounter();
+        } else {
+            logger.warn("purge history reached limit");
```

Comment:
when this happens, u will spawn our logs to death. which will fill the disk even faster, lol.

lower the log level to DEBUG or make it only write this log every 5min or so.

Reply:
Record limit for a day is pretty high so we shouldn't hit the warning in near future. But it's obviously going to hit the limit at one point in future. So debug seems like a better alternative than timer based.

Reply:
i just mean, ur deating the purpose of why u created the limit in the first place.

u created it to not spam the DB in order to prevent the disk from running out of space. but instead, u create an entry in the log file on each message, which gets saved in the log files instead. so you shifted the problem from DB to log file, the disk is still running out of space.